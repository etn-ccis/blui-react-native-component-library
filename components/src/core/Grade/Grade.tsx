/**
 * @format
 * @flow
 */

import React from 'react';
import { Platform, ViewProps } from 'react-native';
import { Avatar } from 'react-native-paper';
import { $DeepPartial } from '@callstack/react-theme-provider';
import { ExtendedTheme, useExtendedTheme } from '@brightlayer-ui/react-native-themes';

/**
 * Props for the Grade component.
 * @typedef {object} GradeProps
 * @prop {string} label - The text that you want to display.
 * @prop {string} [fontColor] - Text color for the Label (Default is theme.colors.onPrimary).
 * @prop {string} [backgroundColor] - Background color of the Label (Default is theme.colors.primary).
 * @prop {number} [size] - The diameter of the circular view.
 * @prop {$DeepPartial<ExtendedTheme>} [theme] - Theme value overrides specific to this component.
 */
export type GradeProps = ViewProps & {
    /**
     * The text shown in the circle
     */
    label: string;
    /**
     * The color of the text label
     * @default theme.colors.onPrimary
     */
    fontColor?: string;
    /**
     * The color of the text label
     * @default theme.colors.primary
     */
    backgroundColor?: string;
    /**
     * The size of the circle in px
     * @default 40
     */
    size?: number;
    /**
     * Theme value overrides
     */
    theme?: $DeepPartial<ExtendedTheme>;
};

export type FixedGradeProps = Omit<GradeProps, 'label' | 'fontColor' | 'backgroundColor'>;

const GradeBase = (props: GradeProps): JSX.Element => {
    const defaultTheme = useExtendedTheme();
    const { label, fontColor, backgroundColor, size = 40, style, theme: themeOverride, ...otherViewProps } = props;

    const theme = useExtendedTheme(themeOverride || props.theme || defaultTheme);

    // Define styles for Avatar.Text component

    const avatarStyle = {
        backgroundColor: backgroundColor || theme.colors.primary,
    };

    // Define styles for text within Avatar.Text
    const textStyle = {
        color: fontColor || theme.colors.onPrimary,
        fontFamily: 'OpenSans-Bold',
        fontWeight: Platform.OS === 'ios' ? ('700' as const) : ('600' as const),
    };

    return (
        <Avatar.Text
            label={label}
            style={[avatarStyle, style]}
            color={fontColor}
            labelStyle={textStyle}
            size={size}
            testID="grade"
            {...otherViewProps}
        />
    );
};

/**
 * A component used to render an A+ Grade
 */
GradeBase.APlus = (props: FixedGradeProps): JSX.Element => (
    <GradeBase label={'A+'} fontColor="#FFFFFF" backgroundColor="#198900" {...props} />
);
/**
 * A component used to render an A Grade
 */
GradeBase.A = (props: FixedGradeProps): JSX.Element => (
    <GradeBase label={'A'} fontColor="#FFFFFF" backgroundColor="#198900" {...props} />
);
/**
 * A component used to render an A- Grade
 */
GradeBase.AMinus = (props: FixedGradeProps): JSX.Element => (
    <GradeBase label={'A-'} fontColor="#FFFFFF" backgroundColor="#64a721" {...props} />
);
/**
 * A component used to render a B+ Grade
 */
GradeBase.BPlus = (props: FixedGradeProps): JSX.Element => (
    <GradeBase label={'B+'} fontColor="#524700" backgroundColor="#afc543" {...props} />
);
/**
 * A component used to render a B Grade
 */
GradeBase.B = (props: FixedGradeProps): JSX.Element => (
    <GradeBase label={'B'} fontColor="#524700" backgroundColor="#FBE365" {...props} />
);
/**
 * A component used to render a B- Grade
 */
GradeBase.BMinus = (props: FixedGradeProps): JSX.Element => (
    <GradeBase label={'B-'} fontColor="#524700" backgroundColor="#f6c543" {...props} />
);
/**
 * A component used to render a C+ Grade
 */
GradeBase.CPlus = (props: FixedGradeProps): JSX.Element => (
    <GradeBase label={'C+'} fontColor="#4B2800" backgroundColor="#f1a821" {...props} />
);
/**
 * A component used to render a C Grade
 */
GradeBase.C = (props: FixedGradeProps): JSX.Element => (
    <GradeBase label={'C'} fontColor="#4B2800" backgroundColor="#ED8B00" {...props} />
);
/**
 * A component used to render a C- Grade
 */
GradeBase.CMinus = (props: FixedGradeProps): JSX.Element => (
    <GradeBase label={'C-'} fontColor="#4B2800" backgroundColor="#dc6508" {...props} />
);
/**
 * A component used to render a D+ Grade
 */
GradeBase.DPlus = (props: FixedGradeProps): JSX.Element => (
    <GradeBase label={'D+'} fontColor="#FFFFFF" backgroundColor="#cb3f11" {...props} />
);
/**
 * A component used to render a D Grade
 */
GradeBase.D = (props: FixedGradeProps): JSX.Element => (
    <GradeBase label={'D'} fontColor="#FFFFFF" backgroundColor="#BA1A1A" {...props} />
);
/**
 * A component used to render a D- Grade
 */
GradeBase.DMinus = (props: FixedGradeProps): JSX.Element => (
    <GradeBase label={'D-'} fontColor="#FFFFFF" backgroundColor="#BA1A1A" {...props} />
);
/**
 * A component used to render an F Grade
 */
GradeBase.F = (props: FixedGradeProps): JSX.Element => (
    <GradeBase label={'F'} fontColor="#FFFFFF" backgroundColor="#9F45F6" {...props} />
);

/**
 * A component used to render a score/grade inside of an Avatar. Includes several sub-components with predefined styles for several common grades.
 */
export const Grade = GradeBase;

/**
 * A component used to render a score/grade inside of an Avatar. Includes several sub-components with predefined styles for several common grades.
 */
export default Grade;
