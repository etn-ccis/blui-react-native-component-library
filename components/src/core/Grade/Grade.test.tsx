import React from 'react';
import { render } from '@testing-library/react-native';
import Grade from './Grade';
import { PaperProvider } from 'react-native-paper';
import { blue } from '@brightlayer-ui/react-native-themes';

describe('Grade.custom', () => {
    it('renders with custom label, color, background color, and size', () => {
        const { toJSON, getByText, getByTestId } = render(
            <Grade label="Custom Grade" fontColor="#FF5733" backgroundColor="#66CCFF" size={50} />
        );

        const avatarElement = getByTestId('grade');
        const avatarStyle = avatarElement.props.style;
        const avatarWidth = avatarStyle[0].width;
        const avatarHeight = avatarStyle[0].height;
        expect(avatarWidth).toBe(50);
        expect(avatarHeight).toBe(50);

        const labelElement = getByText('Custom Grade');
        expect(labelElement).toBeDefined();

        const color = labelElement.props.style[3][1].color;
        expect(color).toBe('#FF5733');

        const backgroundColor = avatarElement.props.style[0].backgroundColor;
        expect(backgroundColor).toBe('#66CCFF');
        expect(toJSON()).toMatchSnapshot();
    });
});

describe('Grade.aPlus', () => {
    it('renders with default label, color, background color, and size', () => {
        const { toJSON, getByText, getByTestId } = render(
            <PaperProvider theme={blue}>
                <Grade.APlus />
            </PaperProvider>
        );

        const avatarElement = getByTestId('grade');
        const avatarStyle = avatarElement.props.style;
        const avatarWidth = avatarStyle[0].width;
        const avatarHeight = avatarStyle[0].height;
        expect(avatarWidth).toBe(40);
        expect(avatarHeight).toBe(40);

        const labelElement = getByText('A+');
        expect(labelElement).toBeDefined();

        const color = labelElement.props.style[3][1].color;
        expect(color).toBe('#FFFFFF');

        const backgroundColor = avatarElement.props.style[0].backgroundColor;
        expect(backgroundColor).toBe('#198900');

        expect(toJSON()).toMatchSnapshot();
    });
});

describe('Grade.a', () => {
    it('renders with default label, color, background color, and size', () => {
        const { toJSON, getByText, getByTestId } = render(
            <PaperProvider theme={blue}>
                <Grade.A />
            </PaperProvider>
        );

        const avatarElement = getByTestId('grade');
        const avatarStyle = avatarElement.props.style;
        const avatarWidth = avatarStyle[0].width;
        const avatarHeight = avatarStyle[0].height;
        expect(avatarWidth).toBe(40);
        expect(avatarHeight).toBe(40);

        const labelElement = getByText('A');
        expect(labelElement).toBeDefined();

        const color = labelElement.props.style[3][1].color;
        expect(color).toBe('#FFFFFF');

        const backgroundColor = avatarElement.props.style[0].backgroundColor;
        expect(backgroundColor).toBe('#198900');
        expect(toJSON()).toMatchSnapshot();
    });
});

describe('Grade.aMinus', () => {
    it('renders with default label, color, background color, and size', () => {
        const { toJSON, getByText, getByTestId } = render(
            <PaperProvider theme={blue}>
                <Grade.AMinus />
            </PaperProvider>
        );

        const avatarElement = getByTestId('grade');
        const avatarStyle = avatarElement.props.style;
        const avatarWidth = avatarStyle[0].width;
        const avatarHeight = avatarStyle[0].height;
        expect(avatarWidth).toBe(40);
        expect(avatarHeight).toBe(40);

        const labelElement = getByText('A-');
        expect(labelElement).toBeDefined();

        const color = labelElement.props.style[3][1].color;
        expect(color).toBe('#FFFFFF');

        const backgroundColor = avatarElement.props.style[0].backgroundColor;
        expect(backgroundColor).toBe('rgba(100, 167, 33, 1)');

        expect(toJSON()).toMatchSnapshot();
    });
});

describe('Grade.bPlus', () => {
    it('renders with default label, color, background color, and size', () => {
        const { toJSON, getByText, getByTestId } = render(
            <PaperProvider theme={blue}>
                <Grade.BPlus />
            </PaperProvider>
        );

        const avatarElement = getByTestId('grade');
        const avatarStyle = avatarElement.props.style;
        const avatarWidth = avatarStyle[0].width;
        const avatarHeight = avatarStyle[0].height;
        expect(avatarWidth).toBe(40);
        expect(avatarHeight).toBe(40);

        const labelElement = getByText('B+');
        expect(labelElement).toBeDefined();

        const color = labelElement.props.style[3][1].color;
        expect(color).toBe('#524700');

        const backgroundColor = avatarElement.props.style[0].backgroundColor;
        expect(backgroundColor).toBe('rgba(176, 197, 68, 1)');

        expect(toJSON()).toMatchSnapshot();
    });
});

describe('Grade.b', () => {
    it('renders with default label, color, background color, and size', () => {
        const { toJSON, getByText, getByTestId } = render(
            <PaperProvider theme={blue}>
                <Grade.B />
            </PaperProvider>
        );

        const avatarElement = getByTestId('grade');
        const avatarStyle = avatarElement.props.style;
        const avatarWidth = avatarStyle[0].width;
        const avatarHeight = avatarStyle[0].height;
        expect(avatarWidth).toBe(40);
        expect(avatarHeight).toBe(40);

        const labelElement = getByText('B');
        expect(labelElement).toBeDefined();

        const color = labelElement.props.style[3][1].color;
        expect(color).toBe('#524700');

        const backgroundColor = avatarElement.props.style[0].backgroundColor;
        expect(backgroundColor).toBe('#FBE365');

        expect(toJSON()).toMatchSnapshot();
    });
});

describe('Grade.bMinus', () => {
    it('renders with default label, color, background color, and size', () => {
        const { toJSON, getByText, getByTestId } = render(
            <PaperProvider theme={blue}>
                <Grade.BMinus />
            </PaperProvider>
        );

        const avatarElement = getByTestId('grade');
        const avatarStyle = avatarElement.props.style;
        const avatarWidth = avatarStyle[0].width;
        const avatarHeight = avatarStyle[0].height;
        expect(avatarWidth).toBe(40);
        expect(avatarHeight).toBe(40);

        const labelElement = getByText('B-');
        expect(labelElement).toBeDefined();

        const color = labelElement.props.style[3][1].color;
        expect(color).toBe('#524700');

        const backgroundColor = avatarElement.props.style[0].backgroundColor;
        expect(backgroundColor).toBe('rgba(246, 198, 68, 1)');

        expect(toJSON()).toMatchSnapshot();
    });
});

describe('Grade.cPlus', () => {
    it('renders with default label, color, background color, and size', () => {
        const { toJSON, getByText, getByTestId } = render(
            <PaperProvider theme={blue}>
                <Grade.CPlus />
            </PaperProvider>
        );

        const avatarElement = getByTestId('grade');
        const avatarStyle = avatarElement.props.style;
        const avatarWidth = avatarStyle[0].width;
        const avatarHeight = avatarStyle[0].height;
        expect(avatarWidth).toBe(40);
        expect(avatarHeight).toBe(40);

        const labelElement = getByText('C+');
        expect(labelElement).toBeDefined();

        const color = labelElement.props.style[3][1].color;
        expect(color).toBe('#4B2800');

        const backgroundColor = avatarElement.props.style[0].backgroundColor;
        expect(backgroundColor).toBe('rgba(242, 168, 33, 1)');

        expect(toJSON()).toMatchSnapshot();
    });
});

describe('Grade.c', () => {
    it('renders with default label, color, background color, and size', () => {
        const { toJSON, getByText, getByTestId } = render(
            <PaperProvider theme={blue}>
                <Grade.C />
            </PaperProvider>
        );

        const avatarElement = getByTestId('grade');
        const avatarStyle = avatarElement.props.style;
        const avatarWidth = avatarStyle[0].width;
        const avatarHeight = avatarStyle[0].height;
        expect(avatarWidth).toBe(40);
        expect(avatarHeight).toBe(40);

        const labelElement = getByText('C');
        expect(labelElement).toBeDefined();

        const color = labelElement.props.style[3][1].color;
        expect(color).toBe('#4B2800');

        const backgroundColor = avatarElement.props.style[0].backgroundColor;
        expect(backgroundColor).toBe('#ED8B00');

        expect(toJSON()).toMatchSnapshot();
    });
});

describe('Grade.cMinus', () => {
    it('renders with default label, color, background color, and size', () => {
        const { toJSON, getByText, getByTestId } = render(
            <PaperProvider theme={blue}>
                <Grade.CMinus />
            </PaperProvider>
        );

        const avatarElement = getByTestId('grade');
        const avatarStyle = avatarElement.props.style;
        const avatarWidth = avatarStyle[0].width;
        const avatarHeight = avatarStyle[0].height;
        expect(avatarWidth).toBe(40);
        expect(avatarHeight).toBe(40);

        const labelElement = getByText('C-');
        expect(labelElement).toBeDefined();

        const color = labelElement.props.style[3][1].color;
        expect(color).toBe('#4B2800');

        const backgroundColor = avatarElement.props.style[0].backgroundColor;
        expect(backgroundColor).toBe('rgba(220, 102, 9, 1)');

        expect(toJSON()).toMatchSnapshot();
    });
});

describe('Grade.dPlus', () => {
    it('renders with default label, color, background color, and size', () => {
        const { toJSON, getByText, getByTestId } = render(
            <PaperProvider theme={blue}>
                <Grade.DPlus />
            </PaperProvider>
        );

        const avatarElement = getByTestId('grade');
        const avatarStyle = avatarElement.props.style;
        const avatarWidth = avatarStyle[0].width;
        const avatarHeight = avatarStyle[0].height;
        expect(avatarWidth).toBe(40);
        expect(avatarHeight).toBe(40);

        const labelElement = getByText('D+');
        expect(labelElement).toBeDefined();

        const color = labelElement.props.style[3][1].color;
        expect(color).toBe('#FFFFFF');

        const backgroundColor = avatarElement.props.style[0].backgroundColor;
        expect(backgroundColor).toBe('rgba(203, 63, 17, 1)');

        expect(toJSON()).toMatchSnapshot();
    });
});

describe('Grade.d', () => {
    it('renders with default label, color, background color, and size', () => {
        const { toJSON, getByText, getByTestId } = render(
            <PaperProvider theme={blue}>
                <Grade.D />
            </PaperProvider>
        );

        const avatarElement = getByTestId('grade');
        const avatarStyle = avatarElement.props.style;
        const avatarWidth = avatarStyle[0].width;
        const avatarHeight = avatarStyle[0].height;
        expect(avatarWidth).toBe(40);
        expect(avatarHeight).toBe(40);

        const labelElement = getByText('D');
        expect(labelElement).toBeDefined();

        const color = labelElement.props.style[3][1].color;
        expect(color).toBe('#FFFFFF');

        const backgroundColor = avatarElement.props.style[0].backgroundColor;
        expect(backgroundColor).toBe('#BA1A1A');

        expect(toJSON()).toMatchSnapshot();
    });
});

describe('Grade.dMinus', () => {
    it('renders with default label, color, background color, and size', () => {
        const { toJSON, getByText, getByTestId } = render(
            <PaperProvider theme={blue}>
                <Grade.DMinus />
            </PaperProvider>
        );

        const avatarElement = getByTestId('grade');
        const avatarStyle = avatarElement.props.style;
        const avatarWidth = avatarStyle[0].width;
        const avatarHeight = avatarStyle[0].height;
        expect(avatarWidth).toBe(40);
        expect(avatarHeight).toBe(40);

        const labelElement = getByText('D-');
        expect(labelElement).toBeDefined();

        const color = labelElement.props.style[3][1].color;
        expect(color).toBe('#FFFFFF');

        const backgroundColor = avatarElement.props.style[0].backgroundColor;
        expect(backgroundColor).toBe('#BA1A1A');

        expect(toJSON()).toMatchSnapshot();
    });
});

describe('Grade.f', () => {
    it('renders with default label, color, background color, and size', () => {
        const { toJSON, getByText, getByTestId } = render(
            <PaperProvider theme={blue}>
                <Grade.F />
            </PaperProvider>
        );

        const avatarElement = getByTestId('grade');
        const avatarStyle = avatarElement.props.style;
        const avatarWidth = avatarStyle[0].width;
        const avatarHeight = avatarStyle[0].height;
        expect(avatarWidth).toBe(40);
        expect(avatarHeight).toBe(40);

        const labelElement = getByText('F');
        expect(labelElement).toBeDefined();

        const color = labelElement.props.style[3][1].color;
        expect(color).toBe('#FFFFFF');

        const backgroundColor = avatarElement.props.style[0].backgroundColor;
        expect(backgroundColor).toBe('#9F45F6');

        expect(toJSON()).toMatchSnapshot();
    });
});
