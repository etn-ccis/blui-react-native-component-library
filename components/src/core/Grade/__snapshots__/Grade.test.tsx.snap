// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grade.a renders with default label, color, background color, and size 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#198900",
        "borderRadius": 20,
        "height": 40,
        "width": 40,
      },
      {
        "alignItems": "center",
        "justifyContent": "center",
      },
      {},
    ]
  }
  testID="grade"
>
  <Text
    numberOfLines={1}
    style={
      [
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "fontFamily": "System",
          "fontWeight": "400",
          "letterSpacing": 0,
        },
        {
          "writingDirection": "ltr",
        },
        [
          {
            "textAlign": "center",
            "textAlignVertical": "center",
          },
          {
            "color": "#FFFFFF",
            "fontSize": 20,
            "lineHeight": 20,
          },
          {
            "color": "#FFFFFF",
            "fontFamily": "OpenSans-Bold",
            "fontWeight": "700",
          },
        ],
      ]
    }
  >
    A
  </Text>
</View>
`;

exports[`Grade.aMinus renders with default label, color, background color, and size 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#64a721",
        "borderRadius": 20,
        "height": 40,
        "width": 40,
      },
      {
        "alignItems": "center",
        "justifyContent": "center",
      },
      {},
    ]
  }
  testID="grade"
>
  <Text
    numberOfLines={1}
    style={
      [
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "fontFamily": "System",
          "fontWeight": "400",
          "letterSpacing": 0,
        },
        {
          "writingDirection": "ltr",
        },
        [
          {
            "textAlign": "center",
            "textAlignVertical": "center",
          },
          {
            "color": "#FFFFFF",
            "fontSize": 20,
            "lineHeight": 20,
          },
          {
            "color": "#FFFFFF",
            "fontFamily": "OpenSans-Bold",
            "fontWeight": "700",
          },
        ],
      ]
    }
  >
    A-
  </Text>
</View>
`;

exports[`Grade.aPlus renders with default label, color, background color, and size 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#198900",
        "borderRadius": 20,
        "height": 40,
        "width": 40,
      },
      {
        "alignItems": "center",
        "justifyContent": "center",
      },
      {},
    ]
  }
  testID="grade"
>
  <Text
    numberOfLines={1}
    style={
      [
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "fontFamily": "System",
          "fontWeight": "400",
          "letterSpacing": 0,
        },
        {
          "writingDirection": "ltr",
        },
        [
          {
            "textAlign": "center",
            "textAlignVertical": "center",
          },
          {
            "color": "#FFFFFF",
            "fontSize": 20,
            "lineHeight": 20,
          },
          {
            "color": "#FFFFFF",
            "fontFamily": "OpenSans-Bold",
            "fontWeight": "700",
          },
        ],
      ]
    }
  >
    A+
  </Text>
</View>
`;

exports[`Grade.b renders with default label, color, background color, and size 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#FBE365",
        "borderRadius": 20,
        "height": 40,
        "width": 40,
      },
      {
        "alignItems": "center",
        "justifyContent": "center",
      },
      {},
    ]
  }
  testID="grade"
>
  <Text
    numberOfLines={1}
    style={
      [
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "fontFamily": "System",
          "fontWeight": "400",
          "letterSpacing": 0,
        },
        {
          "writingDirection": "ltr",
        },
        [
          {
            "textAlign": "center",
            "textAlignVertical": "center",
          },
          {
            "color": "#524700",
            "fontSize": 20,
            "lineHeight": 20,
          },
          {
            "color": "#524700",
            "fontFamily": "OpenSans-Bold",
            "fontWeight": "700",
          },
        ],
      ]
    }
  >
    B
  </Text>
</View>
`;

exports[`Grade.bMinus renders with default label, color, background color, and size 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#f6c543",
        "borderRadius": 20,
        "height": 40,
        "width": 40,
      },
      {
        "alignItems": "center",
        "justifyContent": "center",
      },
      {},
    ]
  }
  testID="grade"
>
  <Text
    numberOfLines={1}
    style={
      [
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "fontFamily": "System",
          "fontWeight": "400",
          "letterSpacing": 0,
        },
        {
          "writingDirection": "ltr",
        },
        [
          {
            "textAlign": "center",
            "textAlignVertical": "center",
          },
          {
            "color": "#524700",
            "fontSize": 20,
            "lineHeight": 20,
          },
          {
            "color": "#524700",
            "fontFamily": "OpenSans-Bold",
            "fontWeight": "700",
          },
        ],
      ]
    }
  >
    B-
  </Text>
</View>
`;

exports[`Grade.bPlus renders with default label, color, background color, and size 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#afc543",
        "borderRadius": 20,
        "height": 40,
        "width": 40,
      },
      {
        "alignItems": "center",
        "justifyContent": "center",
      },
      {},
    ]
  }
  testID="grade"
>
  <Text
    numberOfLines={1}
    style={
      [
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "fontFamily": "System",
          "fontWeight": "400",
          "letterSpacing": 0,
        },
        {
          "writingDirection": "ltr",
        },
        [
          {
            "textAlign": "center",
            "textAlignVertical": "center",
          },
          {
            "color": "#524700",
            "fontSize": 20,
            "lineHeight": 20,
          },
          {
            "color": "#524700",
            "fontFamily": "OpenSans-Bold",
            "fontWeight": "700",
          },
        ],
      ]
    }
  >
    B+
  </Text>
</View>
`;

exports[`Grade.c renders with default label, color, background color, and size 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#ED8B00",
        "borderRadius": 20,
        "height": 40,
        "width": 40,
      },
      {
        "alignItems": "center",
        "justifyContent": "center",
      },
      {},
    ]
  }
  testID="grade"
>
  <Text
    numberOfLines={1}
    style={
      [
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "fontFamily": "System",
          "fontWeight": "400",
          "letterSpacing": 0,
        },
        {
          "writingDirection": "ltr",
        },
        [
          {
            "textAlign": "center",
            "textAlignVertical": "center",
          },
          {
            "color": "#4B2800",
            "fontSize": 20,
            "lineHeight": 20,
          },
          {
            "color": "#4B2800",
            "fontFamily": "OpenSans-Bold",
            "fontWeight": "700",
          },
        ],
      ]
    }
  >
    C
  </Text>
</View>
`;

exports[`Grade.cMinus renders with default label, color, background color, and size 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#dc6508",
        "borderRadius": 20,
        "height": 40,
        "width": 40,
      },
      {
        "alignItems": "center",
        "justifyContent": "center",
      },
      {},
    ]
  }
  testID="grade"
>
  <Text
    numberOfLines={1}
    style={
      [
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "fontFamily": "System",
          "fontWeight": "400",
          "letterSpacing": 0,
        },
        {
          "writingDirection": "ltr",
        },
        [
          {
            "textAlign": "center",
            "textAlignVertical": "center",
          },
          {
            "color": "#4B2800",
            "fontSize": 20,
            "lineHeight": 20,
          },
          {
            "color": "#4B2800",
            "fontFamily": "OpenSans-Bold",
            "fontWeight": "700",
          },
        ],
      ]
    }
  >
    C-
  </Text>
</View>
`;

exports[`Grade.cPlus renders with default label, color, background color, and size 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#f1a821",
        "borderRadius": 20,
        "height": 40,
        "width": 40,
      },
      {
        "alignItems": "center",
        "justifyContent": "center",
      },
      {},
    ]
  }
  testID="grade"
>
  <Text
    numberOfLines={1}
    style={
      [
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "fontFamily": "System",
          "fontWeight": "400",
          "letterSpacing": 0,
        },
        {
          "writingDirection": "ltr",
        },
        [
          {
            "textAlign": "center",
            "textAlignVertical": "center",
          },
          {
            "color": "#4B2800",
            "fontSize": 20,
            "lineHeight": 20,
          },
          {
            "color": "#4B2800",
            "fontFamily": "OpenSans-Bold",
            "fontWeight": "700",
          },
        ],
      ]
    }
  >
    C+
  </Text>
</View>
`;

exports[`Grade.custom renders with custom label, color, background color, and size 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#66CCFF",
        "borderRadius": 25,
        "height": 50,
        "width": 50,
      },
      {
        "alignItems": "center",
        "justifyContent": "center",
      },
      {},
    ]
  }
  testID="grade"
>
  <Text
    numberOfLines={1}
    style={
      [
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "fontFamily": "System",
          "fontWeight": "400",
          "letterSpacing": 0,
        },
        {
          "writingDirection": "ltr",
        },
        [
          {
            "textAlign": "center",
            "textAlignVertical": "center",
          },
          {
            "color": "#FF5733",
            "fontSize": 25,
            "lineHeight": 25,
          },
          {
            "color": "#FF5733",
            "fontFamily": "OpenSans-Bold",
            "fontWeight": "700",
          },
        ],
      ]
    }
  >
    Custom Grade
  </Text>
</View>
`;

exports[`Grade.d renders with default label, color, background color, and size 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#BA1A1A",
        "borderRadius": 20,
        "height": 40,
        "width": 40,
      },
      {
        "alignItems": "center",
        "justifyContent": "center",
      },
      {},
    ]
  }
  testID="grade"
>
  <Text
    numberOfLines={1}
    style={
      [
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "fontFamily": "System",
          "fontWeight": "400",
          "letterSpacing": 0,
        },
        {
          "writingDirection": "ltr",
        },
        [
          {
            "textAlign": "center",
            "textAlignVertical": "center",
          },
          {
            "color": "#FFFFFF",
            "fontSize": 20,
            "lineHeight": 20,
          },
          {
            "color": "#FFFFFF",
            "fontFamily": "OpenSans-Bold",
            "fontWeight": "700",
          },
        ],
      ]
    }
  >
    D
  </Text>
</View>
`;

exports[`Grade.dMinus renders with default label, color, background color, and size 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#BA1A1A",
        "borderRadius": 20,
        "height": 40,
        "width": 40,
      },
      {
        "alignItems": "center",
        "justifyContent": "center",
      },
      {},
    ]
  }
  testID="grade"
>
  <Text
    numberOfLines={1}
    style={
      [
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "fontFamily": "System",
          "fontWeight": "400",
          "letterSpacing": 0,
        },
        {
          "writingDirection": "ltr",
        },
        [
          {
            "textAlign": "center",
            "textAlignVertical": "center",
          },
          {
            "color": "#FFFFFF",
            "fontSize": 20,
            "lineHeight": 20,
          },
          {
            "color": "#FFFFFF",
            "fontFamily": "OpenSans-Bold",
            "fontWeight": "700",
          },
        ],
      ]
    }
  >
    D-
  </Text>
</View>
`;

exports[`Grade.dPlus renders with default label, color, background color, and size 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#cb3f11",
        "borderRadius": 20,
        "height": 40,
        "width": 40,
      },
      {
        "alignItems": "center",
        "justifyContent": "center",
      },
      {},
    ]
  }
  testID="grade"
>
  <Text
    numberOfLines={1}
    style={
      [
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "fontFamily": "System",
          "fontWeight": "400",
          "letterSpacing": 0,
        },
        {
          "writingDirection": "ltr",
        },
        [
          {
            "textAlign": "center",
            "textAlignVertical": "center",
          },
          {
            "color": "#FFFFFF",
            "fontSize": 20,
            "lineHeight": 20,
          },
          {
            "color": "#FFFFFF",
            "fontFamily": "OpenSans-Bold",
            "fontWeight": "700",
          },
        ],
      ]
    }
  >
    D+
  </Text>
</View>
`;

exports[`Grade.f renders with default label, color, background color, and size 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#9F45F6",
        "borderRadius": 20,
        "height": 40,
        "width": 40,
      },
      {
        "alignItems": "center",
        "justifyContent": "center",
      },
      {},
    ]
  }
  testID="grade"
>
  <Text
    numberOfLines={1}
    style={
      [
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "fontFamily": "System",
          "fontWeight": "400",
          "letterSpacing": 0,
        },
        {
          "writingDirection": "ltr",
        },
        [
          {
            "textAlign": "center",
            "textAlignVertical": "center",
          },
          {
            "color": "#FFFFFF",
            "fontSize": 20,
            "lineHeight": 20,
          },
          {
            "color": "#FFFFFF",
            "fontFamily": "OpenSans-Bold",
            "fontWeight": "700",
          },
        ],
      ]
    }
  >
    F
  </Text>
</View>
`;
